src/main/java/woocommerce/utils/ReflectionUtils.java

```diff
@@ -33,14 +33,19 @@ public static ProductVariable jsonStringToProductVariable(String json) {
                 field.setAccessible(true);
                 if (field.isAnnotationPresent(JsonElement.class)) {
                     String key = field.getAnnotation(JsonElement.class).key();
-                    if (field.isAnnotationPresent(JsonElementType.class)) {
-                        // We need to convert String to Integer
-                        Integer prop = null;
-                        if (result.get(field.getAnnotation(JsonElement.class).key()) instanceof String) {
-                            prop = Integer.parseInt((String) result.get(field.getAnnotation(JsonElement.class).key()));
+                    TypeOfJsonElement typeOfElement = field.getAnnotation(JsonElement.class).type();
+
+                    if (typeOfElement != TypeOfJsonElement.NONE) {
+                        //need to cast to specific type
+                        if (typeOfElement == TypeOfJsonElement.INTEGER) {
+                            Integer prop = Integer.parseInt((String) result.get(key));
+                            field.set(productVariable, prop);
+                        } else if (typeOfElement == TypeOfJsonElement.STRING) {
+                            String prop = result.get(key).toString();
```

Comment:
There could be a chance that result.get(key) return null, so toString() method would throw NullPointerException. I think in this case, cast to String is better

Reply:
I agree.